### PR TITLE
Clean up rstar versioned dependencies

### PR DESCRIPTION
**Pull request by nyurik**
Explicitly declare `rstar_0_8` and `rstar_0_9` to make it clearer which version is being used.  For backward compatibility, the `rstar` is now a feature that enables `rstar_0_8`

- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

**Review by lnicola — APPROVED** [review]

File: geo-types/Cargo.toml [hunk 1]
```diff
@@ -11,17 +11,19 @@ description = "Geospatial primitive data types"
 edition = "2021"
 
 [features]
-use-rstar = ["rstar", "approx"]
+# Prefer `use-rstar` feature rather than enabling rstar directly.
+# rstar integration relies on the optional approx crate, but implicit features cannot yet enable other features.
+# See: https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#namespaced-features
```
Comment (michaelkirk) [inline geo-types/Cargo.toml:16]: from https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#namespaced-features

>  Namespaced features has been stabilized in the 1.60 release.

Nice! So in 4 releases (~24 weeks) we could think about utilizing them.

Comment (michaelkirk) [reply geo-types/Cargo.toml:16]: (unrelated to your PR, I know... the diff just caused me to check the link)

**Comment by michaelkirk** [comment]
bors r+

thanks!

**Comment by michaelkirk** [comment]
bors r-

Oh, right, the changelog...

Your GH message seems pretty close to a useful changelog message. I'd recommend proposing something based on that.

**Comment by lnicola** [comment]
Umm, the plan was to drop the plain `rstar` feature in the next minor release, right?

**Comment by michaelkirk** [comment]
> Umm, the plan was to drop the plain rstar feature in the next minor release, right?

Can you clarify what you mean by "minor" - it's a bit confusing because geo-types is 0.x.y

In https://github.com/georust/geo/pull/682#discussion_r757650946 you and @rmanoka discussed it, and it seems like you agreed to remove it in the next *breaking* release.

**Comment by lnicola** [comment]
Next minor as in 0.20.0 (0.19.1 would be a patch release).

**Comment by nyurik** [comment]
My goal for this PR was purely a cleanup without any visible changes. We can later figure out if we want any breaking or non-breaking changes, if anything should be removed/renamed/...

**Comment by nyurik** [comment]
> Your GH message seems pretty close to a useful changelog message. I'd recommend proposing something based on that.

thx @michaelkirk  , done

**Comment by lnicola** [comment]
> What other breaking changes are we planning to merge soon?

I don't think I've checked every open PR, but #751 is breaking AFAICT.

> Can you clarify what you'd like to see happen with this PR @lnicola? I'm a bit confused because you approved it, but now it seems like you're saying you'd prefer not to merge it.

No preference, I was just pointing out that we might have to update this again (and remove it form the changelog) if we merge another breaking change before the next release.

**Comment by nyurik** [comment]
Thx @lnicola , makes sense. The #751 is a bit raw at the moment, so need to merge it just yet until we are clear we really need it. The #760 and #761 are a bit more important to me, as they make it easier to introduce breaking changes as needed, and simplify their reviews.

File: geo-types/CHANGES.md [hunk 2]
```diff
@@ -8,6 +8,8 @@
   * <https://github.com/georust/geo/pull/741>
 * Macros `coord!`, `point!`, `line_string!`, and `polygon!` now support trailing commas such as `coord! { x: 181.2, y: 51.79, }`
   * <https://github.com/georust/geo/pull/752>
+* Internal cleanup: Explicitly declare `rstar_0_8` and `rstar_0_9` features to be explicit which rstar version is being used. For backward compatibility, the `rstar` feature will still enable `rstar_0_8`.
```
Comment (michaelkirk) [inline geo-types/CHANGES.md:11]:
```suggestion
* Internal cleanup: Explicitly declare `use-rstar_0_8` and `use-rstar_0_9` features to be explicit which rstar version is being used. For backward compatibility, the `use-rstar` feature will still enable `use-rstar_0_8`.
```

Comment (michaelkirk) [reply geo-types/CHANGES.md:11]: probably better to talk about the features we want people to type into their Cargo.toml

**Comment by michaelkirk** [comment]
Thanks for clarifying @lnicola - sometimes I have a hard time understanding people on the internet.

**Comment by nyurik** [comment]
Are there any blockers for this noop PR?  Seems like all comments have been resolved?

**Comment by lnicola** [comment]
The changes here are good. I think we should also drop the deprecated feature, but that's a different discussion.

bors r+

**Comment by nyurik** [comment]
@lnicola https://github.com/georust/geo/pull/772

**Comment by nyurik** [comment]
@lnicola I think your bors-foo is not as strong as @michaelkirk 's ;)

**Comment by lnicola** [comment]
https://status.github.com/

**Comment by nyurik** [comment]
So you broke github too?  Nice :)

**Pull request merged**